File: src/main/java/fr/joellehuyen/AirFrance_FlightReviews/models/Review.java
package fr.joellehuyen.AirFrance_FlightReviews.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Review {
    @Id
    @GeneratedValue(strategy= GenerationType.UUID)
    private String id;

    @ManyToOne
    @JoinColumn(name = "flight_id", nullable=false)
    private Flight flight;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable=false)
    private User user;

    private int rating;
    private String comments;
    private LocalDateTime reviewDate;
    private ReviewStatus status;
    private String response;

    public Review(User user, Flight flight, int rating, String comments) {
        this.user = user;
        this.flight = flight;
        this.rating = rating;
        this.comments = comments;
        this.reviewDate = LocalDateTime.now();
        this.status = ReviewStatus.PENDING;
    }
}

File: src/app/data/review.ts
import {Flight} from "./flight";
import {User} from "./user";

export interface Review {
  reviewId: string;
  flight:Flight;
  rating: number;
  comments: string;
  user: User;
  reviewDate :Date;
  reviewStatus:string;
}

export type ReviewCreateInput = Omit<Review, 'reviewId' | 'user' | 'flight'>&{
  flightId: string;
  userId: string;
}

